Packet++/src/X509ExtensionDataDecoder.cpp

```diff
@@ -0,0 +1,289 @@
+#include <unordered_map>
+
+namespace pcpp
+{
+	std::string X509ExtendedKeyUsagePurpose::toString() const
+	{
+		switch (m_Value)
+		{
+		case ServerAuth:
+			return "ServerAuth";
+		case ClientAuth:
+			return "ClientAuth";
+		case CodeSigning:
+			return "CodeSigning";
+		case EmailProtection:
+			return "EmailProtection";
+		case TimeStamping:
+			return "TimeStamping";
+		case OCSPSigning:
+			return "OCSPSigning";
+		case IPSecEndSystem:
+			return "IPSecEndSystem";
+		case IPSecTunnel:
+			return "IPSecTunnel";
+		case IPSecUser:
+			return "IPSecUser";
+		case AnyExtendedKeyUsage:
+			return "AnyExtendedKeyUsage";
+		case SmartCardLogon:
+			return "SmartCardLogon";
+		case EncryptedFileSystem:
+			return "EncryptedFileSystem";
+		case DocumentSigning:
+			return "DocumentSigning";
+		default:
+			return "Unknown";
+		}
+	}
+
+	std::string X509ExtendedKeyUsagePurpose::getOidValue() const
+	{
+		switch (m_Value)
+		{
+		case ServerAuth:
+			return "1.3.6.1.5.5.7.3.1";
+		case ClientAuth:
+			return "1.3.6.1.5.5.7.3.2";
+		case CodeSigning:
+			return "1.3.6.1.5.5.7.3.3";
+		case EmailProtection:
+			return "1.3.6.1.5.5.7.3.4";
+		case TimeStamping:
+			return "1.3.6.1.5.5.7.3.8";
+		case OCSPSigning:
+			return "1.3.6.1.5.5.7.3.9";
+		case IPSecEndSystem:
+			return "1.3.6.1.5.5.7.3.5";
+		case IPSecTunnel:
+			return "1.3.6.1.5.5.7.3.6";
+		case IPSecUser:
+			return "1.3.6.1.5.5.7.3.7";
+		case AnyExtendedKeyUsage:
+			return "2.5.29.37.0";
+		case SmartCardLogon:
+			return "1.3.6.1.4.1.311.20.2.2";
+		case EncryptedFileSystem:
+			return "1.3.6.1.4.1.311.10.3.4";
+		case DocumentSigning:
+			return "1.3.6.1.4.1.311.10.3.12";
+		default:
+			return "0.0";
+		}
+	}
+
+	static const std::unordered_map<std::string, X509ExtendedKeyUsagePurpose::Value>
+	    X509ExtendedKeyUsagePurposeOidMap = {
+		    { "1.3.6.1.5.5.7.3.1",       X509ExtendedKeyUsagePurpose::ServerAuth          },
+		    { "1.3.6.1.5.5.7.3.2",       X509ExtendedKeyUsagePurpose::ClientAuth          },
+		    { "1.3.6.1.5.5.7.3.3",       X509ExtendedKeyUsagePurpose::CodeSigning         },
+		    { "1.3.6.1.5.5.7.3.4",       X509ExtendedKeyUsagePurpose::EmailProtection     },
+		    { "1.3.6.1.5.5.7.3.8",       X509ExtendedKeyUsagePurpose::TimeStamping        },
+		    { "1.3.6.1.5.5.7.3.9",       X509ExtendedKeyUsagePurpose::OCSPSigning         },
+		    { "1.3.6.1.5.5.7.3.5",       X509ExtendedKeyUsagePurpose::IPSecEndSystem      },
+		    { "1.3.6.1.5.5.7.3.6",       X509ExtendedKeyUsagePurpose::IPSecTunnel         },
+		    { "1.3.6.1.5.5.7.3.7",       X509ExtendedKeyUsagePurpose::IPSecUser           },
+		    { "2.5.29.37.0",             X509ExtendedKeyUsagePurpose::AnyExtendedKeyUsage },
+		    { "1.3.6.1.4.1.311.20.2.2",  X509ExtendedKeyUsagePurpose::SmartCardLogon      },
+		    { "1.3.6.1.4.1.311.10.3.4",  X509ExtendedKeyUsagePurpose::EncryptedFileSystem },
+		    { "1.3.6.1.4.1.311.10.3.12", X509ExtendedKeyUsagePurpose::DocumentSigning     },
```

Comment:
missing `0.0`?

Reply:
Fixed in https://github.com/seladb/PcapPlusPlus/pull/1882/commits/b840760535d63eac8762aceb55626252c432e8e1